k8s-test-runner/README.md

```diff
@@ -0,0 +1,144 @@
+## Preparing to Run Tests on Staging
+
+Ensure you complete the following steps before executing tests on the staging environment:
+
+1. **Connect to the VPN**
+
+2. **AWS Login with Staging Profile**
+
+   Authenticate to AWS using your staging profile, specifically with the `StagingEKSAdmin` role. Execute the following command:
+
+   ```sh
+   aws sso login --profile staging
+   ```
+
+3. **Verify Authorization**
+
+   Confirm your authorization status by listing the namespaces in the staging cluster. Run `kubectl get namespaces`. If you see a list of namespaces, this indicates successful access to the staging cluster.
+
+## Running Tests
+
+### Creating an Image with the Test Binary
+
+Before running tests, you must create a Docker image containing the test binary. To do this, execute the `create-test-image` command and provide the path to the test folder you wish to package. This command:
+
+1. Compiles test binary under `<path-to-test-folder>`
+2. Creates a docker image with the test binary
+3. Pushes the docker image to the image registry (e.g. Staging ECR)
+
+```sh
+go run ./cmd/main.go create-test-image --image-registry-url <staging-ecr-registry-url> --image-tag "<image-tag>" "<path-to-test-folder>"
+```
+
+Where `image-tag` should be a descriptive name for your test, such as "mercury-load-tests".
+
+### Running the Test in Kubernetes
+
+If a Docker image containing the test binary is available in an image registry (such as staging ECR), use `run` command to execute the test in K8s.
+
+```
+go run ./cmd/main.go run -c "<path-to-test-runner-toml-config>"
+```
+
+The TOML config should specify the test runner configuration as follows:
+
+```
+namespace = "e2e-tests"
+rbac_role_name = "" # RBAC role name for the chart 
```

Comment:
Maybe we should add a default role name here, it should still be secure but will be problematic for QAs to figure it out?